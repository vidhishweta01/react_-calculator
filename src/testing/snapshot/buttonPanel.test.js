import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../../component/buttonPanel';

describe('ButtonPanel component', () => {
  it('matches the snapshot', () => {
    const handleClick = () => 'just for testing';
    const panel = renderer.create(<ButtonPanel clickHandler={handleClick} />).toJSON();
    expect(panel).toMatchSnapshot();
  });
});
