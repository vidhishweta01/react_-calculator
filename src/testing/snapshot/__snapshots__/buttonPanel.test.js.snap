// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel component matches the snapshot 1`] = `
<div
  className="mainCalc"
>
  <div
    className="button_row"
  >
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange normal"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="button_row"
  >
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange normal"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="button_row"
  >
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange normal"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="button_row"
  >
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange normal"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="button_row"
  >
    <button
      className="button gray double"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button gray normal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange normal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
